src/visualization/visalizeEpistasis/findSubsystemOfGenes.m

```diff
@@ -0,0 +1,153 @@
+% uSys:    unique subsys
+%
+% Written by:
+% Chintan Joshi when at CSU in 2014. See figures in following publication:
+% Joshi CJ and Prasad A, 2014, "Epistatic interactions among metabolic genes
+% depend upon environmental conditions", Mol. BioSyst., 10, 2578� 2589.
+
+if (nargin<2)
+    genes=model.genes;
+end
+if (nargin<3)
+    pseudoFlag=false;
+end
+
+for i=1:length(genes)
+    % find the reactions that were constrained by this gene
+    [cRxns,iRxns]=findRxnsConstrained(model,genes(i,1),false);
+    if pseudoFlag==false
+        rxns{i,1}=iRxns{1,1};
+        rxnNames{i,1} = model.rxnNames(ismember(model.rxns,iRxns{1,1}));
+    else
+        if (~isempty(cRxns{1,1}))
+            rxns{i,1}=cRxns{1,1};
+            rxnNames{i,1} = model.rxnNames(ismember(model.rxns,cRxns{1,1}));
+        else
+            rxns{i,1}=iRxns{1,1};
+            rxnNames{i,1} = model.rxnNames(ismember(model.rxns,iRxns{1,1}));
+        end
+    end
+    for j=1:length(rxns{i,1})
+        rxnInds=find(strcmp(rxns{i,1}{j,1},model.rxns)==1); % find the index of each of the reactions involved
+    end
+%     if ~isempty(rxnInds)
+    subs{i,1}=model.subSystems{rxnInds,1}; % finds the subsytem of the reaction
+%     end
+%     rxnInds=[];
+    if length(unique(subs{i,1}))==1 % checks if all the reactions belong to the same compartment
+        subsys{i,1}=unique(subs{i,1}); % if they do, stores them under one string
+    else
+        subsys{i,1}=subs(i,1); % else, stores them as a second-ordered cell string
+    end
+end
+
+uSys=unique(convertMyCell2List(subsys,2));
+for i=1:length(uSys)
+    c=0;
+    geneIdx=0;
+    for j=1:length(subsys)
+        indc=find(strcmp(uSys{i,:},subsys{j,1})==1);
+        if (~isempty(indc))
+            c=c+1;
+            geneIdx(c,1)=j;
+        end
+    end
+    subsysGenes{i,1}=genes(geneIdx,1);
+end
+for i=1:length(subsysGenes)
+Nr(i,1)=length(subsysGenes{i,1});
+end
+% Nrp=Nr/sum(Nr); % if wanting to normalize or get frequency insteady.
+[Nr,ind] = sort(Nr,'ascend');
+uComp4plot = uSys(ind);
+barh(Nr);
+set(gca,'YAxisLocation','Right');
+set(gca,'YTickLabelMode','auto');
+set(gca,'YTickMode','auto');
+set(gca,'YTick',1:1:length(Nr));
+set(gca,'YTickLabel',uComp4plot);
+
+function B = convertMyCell2List(A,dimSense)
+
+% This function linearizes a cell array if some of the cells in the array are another embbedded cell arrays.
+% (hence, two degrees of cell array)
+
+% This will only work for atmost two degrees of cell array.
+
+% A=cell array to be linearized.
+% dimSense=determines if inner cells are row (dimSense=1) vectors or columns (dimSense=2, default)
+
+cnt=0;
+if nargin<2
+    dimSense=2;
+end
+for i=1:length(A)
+    for j=1:length(A{i,1})
+        cnt=cnt+1;
+        if dimSense==1
+            B{cnt,1}=A{i,1}{1,j};
+        elseif dimSense==2
+            B{cnt,1}=A{i,1}{j,1};
+        end
+    end
+end
+
+function [constrRxns,invRxns,rxns]=findRxnsConstrained(model,genes,verbFlag)
```

Comment:
What do you intend to achieve with this function?
At the moment, invRxns and constrRxns have very different meanings.
invRxns contains lists of all associated reactions (i.e. just present in the GPR rule) per gene, while constrRxns contains those reactions, which, if the gene is inactivated, are sufficient to make their GPR evaluate to false.
If you want invRxns to represent the set of reactions fr which this gene alone is sufficient I would suggest using `findRxnsActiveWithGenes`.